admin/src/app/header/header.component.ts

```diff
@@ -1,49 +1,67 @@
+  animate,
+  state,
+  style,
+  transition,
+  trigger,
+} from "@angular/animations";
+import { Component, OnInit } from "@angular/core";
+import { Router } from "@angular/router";
 import { User } from "@utility/security/user";
-import { ConfigService } from '@utility/services/config.service';
-import { KeycloakService } from '../../core/services/keycloak.service';
+import { ConfigService } from "@utility/services/config.service";
+// import { KeycloakService } from "../../core/services/keycloak.service";
+import { CognitoService } from "../../core/services/cognito.service";
 
 @Component({
-  selector: 'app-header',
-  templateUrl: './header.component.html',
-  styleUrls: ['./header.component.scss'],
+  selector: "app-header",
+  templateUrl: "./header.component.html",
+  styleUrls: ["./header.component.scss"],
   animations: [
-    trigger('toggleNav', [
-      state('navClosed', style({height: '0'})),
-      state('navOpen', style({height: '*'})),
-      transition('navOpen => navClosed', [animate('0.2s')]),
-      transition('navClosed => navOpen', [animate('0.2s')])
-    ])
-  ]
+    trigger("toggleNav", [
+      state("navClosed", style({ height: "0" })),
+      state("navOpen", style({ height: "*" })),
+      transition("navOpen => navClosed", [animate("0.2s")]),
+      transition("navClosed => navOpen", [animate("0.2s")]),
+    ]),
+  ],
 })
 export class HeaderComponent implements OnInit {
-  isNavMenuOpen = true; 
+  isNavMenuOpen = true;
   environmentDisplay: string;
   logoutMsg: string = "Logout";
   user: User;
 
-  constructor(private keycloakService: KeycloakService, private configService: ConfigService, public router: Router) {
+  constructor(
+    // private keycloakService: KeycloakService,
+    private configService: ConfigService,
+    public router: Router,
+    private cognitoService: CognitoService
+  ) {
     this.environmentDisplay = configService.getEnvironmentDisplay();
-    this.user = this.keycloakService.getUser();
+    // this.user = this.keycloakService.getUser();
+    this.user = this.cognitoService.getUser();
     if (this.user) {
-      this.logoutMsg += ' ' + this.user.displayName;
+      this.logoutMsg += " " + this.user.displayName;
     }
   }
 
   ngOnInit() {
     if (!this.user || !this.user.isAuthorizedForAdminSite()) {
       // If on not-authorized page, or if just logged out, don't redirect to not-authorized page as would cause an infinite loop.
-      if (window.location.href.indexOf('/not-authorized') == -1 && window.location.href.indexOf("loggedout=true") == -1) {
-        this.router.navigate(['/not-authorized']);
+      if (
+        window.location.href.indexOf("/not-authorized") == -1 &&
+        window.location.href.indexOf("loggedout=true") == -1
+      ) {
+        this.router.navigate(["/not-authorized"]);
       }
     }
   }
 
-  navigateToLogout() {
-    this.keycloakService.logout();
-    window.location.href = this.keycloakService.getLogoutURL();
+  async navigateToLogout() {
```

Comment:
I'm not sure if there are any consequences of changing this to async. Just needs testing to make sure it works the same.

Reply:
This method is used from on clicked (html tag).
Angular can take care of async.
Changing to "async" is mainly for the "await" new Cognito's setup to logout.
Verified and it works.